Adding Euler Bend Capability

ejonkerUBC opened the pull request:
**waveguide_gui.ui**
-	Added ‘bend_type’ field which can either be specified as ‘bezier’ or ‘euler’
o	‘bezier’ is the default/backwards compatible parameter for old ‘WAVEGUIDES.xml’ files
o	‘euler’ is a new option that will create euler bends
-	Changed ‘bezier’ parameter to ‘bend_parameter’ which can be either the bezier parameter or the p-parameter for euler bends

![image](https://github.com/user-attachments/assets/213c116c-313a-41ac-bd52-22580ca4031b)
 
**siepic_tools > core.py > WaveguideGUI( ) class**
config_changed( )
-	Added capability to read ‘bend_type’ from waveguide GUI
o	If ‘bend_type’ is not specified, the type is set to ‘bezier’
o	If ‘adiabatic’ is false, set ‘bend_parameter’ to an empty string

get_parameters( )
-	Added ‘bend_type’ to ‘params’ dictionary
-	Changed ‘bezier’ to ‘bend_parameter’ in ‘params’ dictionary

**utils > init.py**
load_waveguides_by_tech( )
-	If ‘bend_parameter’ is not a given field, but ‘bezier’ is, set ‘bend_parameter’ = ‘bezier’ (for reverse compatability)
-	if bend_parameter is specified, also set ‘bezier’ to ‘bend_parameter’ to avoid unforeseen compatibility issues with anything else that looks for ‘bezier’

euler_bend( )
-	Created euler bend function that takes p parameter and r as inputs and creates an euler bend with effective radius r. Returns a set of points that trace out the curve like the ‘arc_bezier’ function

**layout.py**
layout_waveguide3( )
-	If no ‘bend_parameter’ is specified, adiabatic is set to false and bend_parameter is set to an empty sting 
-	Added bend_parameter and bend_type to inputs of layout_waveguide2() function call

layout_waveguide2()
-	Added ‘bend_parameter’ and ‘bend_type’ to inputs
-	If adiabatic and euler is specified, create euler bend
-	If adiabatic and bezier is specified, create Bezier bend
 
![image](https://github.com/user-attachments/assets/e0bb762a-e4b4-4a93-893c-daba0a78dafe)

ejonkerUBC committed:
Add files via upload

ejonkerUBC committed:
Add files via upload

ejonkerUBC committed:
Add files via upload

ejonkerUBC committed:
Add files via upload

ejonkerUBC committed:
Add files via upload

ejonkerUBC committed:
Add files via upload

lukasc-ubc commented:
- siepic_tools > core.py > WaveguideGUI( ) class:  we should make the "Waveguide Specification" fields read-only, since that's what they are...
- Could you make the Bend Type a Dropdown, rather than raw text?  (but if it is read-only, maybe then text is fine?)
- Make it a separate row
- I suggest adding "circular" as an option (regular 90º, non-adiabatic)
- maybe capitalization?  "Bezier" "Euler" since these are named after people
- Remove the check mark for "adiabatic" since Bend Type will take care of that.
- it would be nice to improve the GUI window, to make it wider so that the full text is visible

lukasc-ubc commented:
@ejonkerUBC

ejonkerUBC commented:
Yep Ill work on the changes. I just haven't got around to it yet

ejonkerUBC committed:
Made requested changes

ejonkerUBC commented:
![Screenshot 2025-05-27 152719](https://github.com/user-attachments/assets/bfb5f93c-9b1f-467b-89d2-ae27f29914ac)

Finally got around to making the changes,
- GUI is updated per requests
   - widened window
   - removed 'adiabatic' check box
   - made specs read only
   - changed um to μm
   - separated param and bend rows
- capitalized Bezier and Euler
